lib/core/test/element_histogram_test.cpp

```diff
@@ -65,3 +65,69 @@ TEST(ElementHistogramTest, no_variance) {
                      edges);
   EXPECT_EQ(result_vals, std::vector<double>({20 + 30, 40 + 50}));
 }
+
+TEST(ElementHistogramTest, nan_values_are_dropped) {
+  std::vector<double> edges{0, 4, 6};
+  std::vector<double> events{1, std::nan("2"), 3, 4, 5, 6, 7};
+  std::vector<double> weight_vals{10, 20, 30, 40, 50, 60, 70};
+  std::vector<double> weight_vars{100, 200, 300, 400, 500, 600, 700};
+  std::vector<double> result_vals{0, 0};
+  std::vector<double> result_vars{0, 0};
+  element::histogram(
+      ValueAndVariance(scipp::span(result_vals), scipp::span(result_vars)),
+      events,
+      ValueAndVariance(scipp::span(weight_vals), scipp::span(weight_vars)),
+      edges);
+  EXPECT_EQ(result_vals, std::vector<double>({10 + 30, 40 + 50}));
+  EXPECT_EQ(result_vars, std::vector<double>({100 + 300, 400 + 500}));
+}
+
+TEST(ElementHistogramTest, nan_values_are_dropped_linspace_bins) {
+  std::vector<double> edges{0, 2, 4, 6};
+  std::vector<double> events{1, std::nan("2"), 3, 4, 5, 6, 7};
+  std::vector<double> weight_vals{10, 20, 30, 40, 50, 60, 70};
+  std::vector<double> weight_vars{100, 200, 300, 400, 500, 600, 700};
+  std::vector<double> result_vals{0, 0, 0};
+  std::vector<double> result_vars{0, 0, 0};
+  element::histogram(
+      ValueAndVariance(scipp::span(result_vals), scipp::span(result_vars)),
+      events,
+      ValueAndVariance(scipp::span(weight_vals), scipp::span(weight_vars)),
+      edges);
+  EXPECT_EQ(result_vals, std::vector<double>({10, 30, 40 + 50}));
+  EXPECT_EQ(result_vars, std::vector<double>({100, 300, 400 + 500}));
+}
+
+TEST(ElementHistogramTest, infinite_values_are_dropped) {
+  std::vector<double> edges{0, 4, 6};
+  std::vector<double> events{std::numeric_limits<double>::infinity(),  2, 3, 4,
+                             -std::numeric_limits<double>::infinity(), 6, 7};
```

Comment:
This should have passed before this change, no?

Reply:
Yes, you're right, they pass also on `main`. Should I leave them in anyway?

Reply:
Yes, unless such a test exists already.